libraries/AP_Scripting/examples/ahrs-source-gps-optflow.lua

```diff
@@ -0,0 +1,207 @@
+-- SCR_USER3 holds the threshold for optical flow quality (about 50 is a good choice)
+-- SCR_USER4 holds the threshold for optical flow innovations (about 0.15 is a good choice)
+--
+-- When the 2nd auxiliary switch (300/Scripting1) is pulled high automatic source selection uses these thresholds:
+
+local rangefinder_rotation = 25     -- check downward (25) facing lidar
+local source_prev = 0               -- previous source, defaults to primary source
+local sw_source_prev = -1           -- previous source switch position
+local sw_auto_pos_prev = -1         -- previous auto source switch position
+local auto_switch = false           -- true when auto switching between sources is active
+local gps_usable_accuracy = 1.0     -- GPS is usable if speed accuracy is at or below this value
+local vote_counter_max = 20         -- when a vote counter reaches this number (i.e. 2sec) source may be switched
+local gps_vs_opticalflow_vote = 0   -- vote counter for GPS vs optical (-20 = GPS, +20 = optical flow)
+local scr_user1_param = Parameter() -- user1 param (rangefinder altitude threshold)
+local scr_user2_param = Parameter() -- user2 param (GPS speed accuracy threshold)
+local scr_user3_param = Parameter() -- user3 param (optical flow quality threshold)
+local scr_user4_param = Parameter() -- user4 param (optical flow innovation threshold)
+
+-- initialise parameters
+assert(scr_user1_param:init('SCR_USER1'), 'ahrs-source.lua: could not find SCR_USER1 parameter')
+assert(scr_user2_param:init('SCR_USER2'), 'ahrs-source.lua: could not find SCR_USER2 parameter')
+assert(scr_user3_param:init('SCR_USER3'), 'ahrs-source.lua: could not find SCR_USER3 parameter')
+assert(scr_user4_param:init('SCR_USER4'), 'ahrs-source.lua: could not find SCR_USER4 parameter')
+
+-- play tune on buzzer to alert user to change in active source set
+function play_source_tune(source)
+  if (source) then
+    if (source == 0) then
+      notify:play_tune("L8C")       -- one long lower tone
+    elseif (source == 1) then
+      notify:play_tune("L12DD")     -- two fast medium tones
+    elseif (source == 2) then
+      notify:play_tune("L16FFF")    -- three very fast, high tones
+    end
+  end
+end
+
+-- the main update function
+function update()
+
+  -- check switches are configured
+  -- source selection from RCx_FUNCTION = 90 (EKF Source Select)
+  -- auto source from RCx_FUNCTION = 300 (Scripting1)
+  local rc_function_source = rc:find_channel_for_option(90)
+  local rc_function_auto = rc:find_channel_for_option(300)
+  if (rc_function_source == nil) or (rc_function_auto == nil) then
+    gcs:send_text(0, "ahrs-source.lua: RCx_FUNCTION=90 or 300 not set!")
+    return update, 1000
+  end
+
+  -- check rangefinder distance threshold has been set
+  local rangefinder_thresh_dist = scr_user1_param:get()     -- SCR_USER1 holds rangefinder threshold
+  if (rangefinder_thresh_dist == nil) or (rangefinder_thresh_dist <= 0) then
```

Comment:
I don't usually bother checking the return of the gets, the only way this will fail is if the pointer is somehow lost. If that happens it will be broken until you re-init, so you might as well just error out on trying to do the comparison on a nil.

Reply:
ok, removed the nil check on the param get.